### PR TITLE
[UX] default to minimal logging (no module/line number/timestamp).

### PR DESCRIPTION
**Pull request by concretevitamin**
`sky launch --gpus T4`

Before:
Screen Shot 2024-09-24 at 13 46 56

After:

Screen Shot 2024-09-24 at 13 46 22

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: sky/utils/env_options.py [hunk 1]
```diff
@@ -5,17 +5,27 @@
...
+    MINIMIZE_LOGGING = ('SKYPILOT_MINIMIZE_LOGGING', True)
     # Internal: this is used to skip the cloud user identity check, which is
     # used to protect cluster operations in a multi-identity scenario.
     # Currently, this is only used in the job and serve controller, as there
     # will not be multiple identities, and skipping the check can increase
     # robustness.
-    SKIP_CLOUD_IDENTITY_CHECK = 'SKYPILOT_SKIP_CLOUD_IDENTITY_CHECK'
+    SKIP_CLOUD_IDENTITY_CHECK = ('SKYPILOT_SKIP_CLOUD_IDENTITY_CHECK', False)
+
+    def __init__(self, env_var, default):
```
Comment (cblmemo) [inline sky/utils/env_options.py:21]:
```suggestion
    def __init__(self, env_var: str, default: bool) -> None:
```

File: sky/utils/env_options.py [hunk 2]
```diff
@@ -5,17 +5,27 @@
...
-    IS_DEVELOPER = 'SKYPILOT_DEV'
-    SHOW_DEBUG_INFO = 'SKYPILOT_DEBUG'
-    DISABLE_LOGGING = 'SKYPILOT_DISABLE_USAGE_COLLECTION'
-    MINIMIZE_LOGGING = 'SKYPILOT_MINIMIZE_LOGGING'
+
+    # (env var name, default value)
+    IS_DEVELOPER = ('SKYPILOT_DEV', False)
+    SHOW_DEBUG_INFO = ('SKYPILOT_DEBUG', False)
+    DISABLE_LOGGING = ('SKYPILOT_DISABLE_USAGE_COLLECTION', False)
+    MINIMIZE_LOGGING = ('SKYPILOT_MINIMIZE_LOGGING', True)
```
Comment (Michaelvll) [inline sky/utils/env_options.py:13]: We should show still show the stamps and file name/lines when `SKYPILOT_DEBUG` is set.

Comment (concretevitamin) [reply sky/utils/env_options.py:13]: Good catch, done. PTAL.

**Comment by Michaelvll** [comment]
Ahh, forgot to send the comment. Can we make sure that `provision.log` still get the stamps? It should be done to have different handlers for our logging module for stdout and filehandler.

**Pull request merged**